src/libpanic_unwind/lib.rs

```diff
@@ -71,6 +71,7 @@ mod windows;
 // hairy and tightly coupled, for more information see the compiler's
 // implementation of this.
 #[no_mangle]
+#[allow(improper_ctypes)]
 pub unsafe extern "C" fn __rust_maybe_catch_panic(f: fn(*mut u8),
```

Comment:
I wonder why this even has an `extern "C"` ABI, if it's used for Rust -> Rust calls (`panic_impl` uses the Rust ABI for example).

Reply:
I think this should be addressed in a follow-up.